tools/getrealdeps.rb

```diff
@@ -1,121 +1,141 @@
+  pkgdeps = pkgdepsfiles.map do |file|
+              whatprovidesfxn(file, pkg)
+            end.sort.reject { |i| i.include?(pkg) }.map { |i| i.split("\n")}.flatten.uniq
+
+  # Massage the glibc entries in the dependency list.
+  pkgdeps = pkgdeps.map { |i| i.gsub(/glibc_build.*/, 'glibc') }.uniq
+  pkgdeps = pkgdeps.map { |i| i.gsub(/glibc_lib.*/, 'glibc_lib') }.uniq.map(&:strip).reject(&:empty?)
+
+  # Look for missing runtime dependencies.
+  missingpkgdeps = pkgdeps.reject { |i| File.read("#{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb").include?("depends_on '#{i}'") unless File.read("#{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb").include?("depends_on '#{i}' => :build") }
 
-if missingpkgdeps.length.positive?
-  puts "\nPackage file #{pkg}.rb is missing these runtime library dependencies:"
-  missingpkgdeps.each do |i|
+  puts "\nPackage #{pkg} has runtime library dependencies on these packages:"
+  pkgdeps.each do |i|
     puts "  depends_on '#{i}' # R"
   end
+
+  if missingpkgdeps.length.positive?
+    puts "\nPackage file #{pkg}.rb is missing these runtime library dependencies:"
+    missingpkgdeps.each do |i|
+      puts "  depends_on '#{i}' # R"
+    end
+  end
+
+  # Add missing dependencies to the package.
+  missingpkgdeps.each do |adddep|
+    puts "\n Adding deps: #{adddep}"
+    gawk_cmd = if File.foreach("#{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb").grep(/depends_on/).any?
+      # This files contains dependencies already, so add new deps after existing dependencies.
+                 "gawk -i inplace -v dep=\"  depends_on '#{adddep}' # R\" 'FNR==NR{ if (/depends_on/) p=NR; next} 1; FNR==p{ print dep }' #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb"
+               else
+      # This files doesn't contain deps, so just add new deps.
+                 "gawk -i inplace -v dep=\"  depends_on '#{adddep}' # R\" 'FNR==NR{ if (/})/) p=NR; next} 1; FNR==p{ print \"\\n\" dep }' #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb"
+               end
+    system(gawk_cmd)
+  end
 end
 
-# Add missing dependencies to the package.
-missingpkgdeps.each do |adddep|
-  puts "\n Adding deps: #{adddep}"
-  gawk_cmd = if File.foreach("#{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb").grep(/depends_on/).any?
-    # This files contains dependencies already, so add new deps after existing dependencies.
-               "gawk -i inplace -v dep=\"  depends_on '#{adddep}' # R\" 'FNR==NR{ if (/depends_on/) p=NR; next} 1; FNR==p{ print dep }' #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb"
-             else
-    # This files doesn't contain deps, so just add new deps.
-               "gawk -i inplace -v dep=\"  depends_on '#{adddep}' # R\" 'FNR==NR{ if (/})/) p=NR; next} 1; FNR==p{ print \"\\n\" dep }' #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb #{CREW_PREFIX}/lib/crew/packages/#{pkg}.rb"
-             end
-  system(gawk_cmd)
+ARGV.each do |package|
```

Comment:
How does the script handle no arguments?

Reply:
That's handled at the top of the script.
```
 $ tools/getrealdeps.rb
Getrealdeps checks for the runtime dependencies of a package.
The runtime dependencies are added if the package file is missing them.
Usage: getrealdeps.rb [--use_crew_dest_dir] <packagename>
```